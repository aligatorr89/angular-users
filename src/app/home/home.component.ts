import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.css']
})
export class HomeComponent implements OnInit {
  title: string = 'angular-users';
  someNumber: number = 1;

  constructor() { }

  ngOnInit() {
  }

  click() {
    this.someNumber++;
  }

}
