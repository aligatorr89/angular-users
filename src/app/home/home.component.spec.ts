import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { DebugElement } from '@angular/core';
import { By } from "@angular/platform-browser";

import { HomeComponent } from './home.component';

describe('HomeComponent', () => {
  let component: HomeComponent;
  let fixture: ComponentFixture<HomeComponent>;
  let componentDomDebug: DebugElement;
  let componentDom: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ HomeComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HomeComponent);
    component = fixture.componentInstance;
    componentDomDebug = fixture.debugElement;
    componentDom = componentDomDebug.nativeElement;
    fixture.detectChanges();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test(`should have property title 'angular-users'`, () => {
    expect(component.title).toEqual('angular-users');
  });

  test(`should have property someNumber`, () => {
    expect(component.someNumber).toBe(1);
  });

  test('should render title in a h1 tag', () => {
    expect(componentDomDebug.nativeElement.querySelector('h1').textContent).toContain('Welcome to angular-users!');
  });

  test('should have logo', () => {
    const element = componentDomDebug.nativeElement.querySelector('img[src^="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNTAgMjUwIj4KICAgIDxwYXRoIGZpbGw9IiNERDAwMzEiIGQ9Ik0xMjUgMzBMMzEuOSA2My4ybDE0LjIgMTIzLjFMMTI1IDIzMGw3OC45LTQzLjcgMTQuMi0xMjMuMXoiIC8+CiAgICA8cGF0aCBmaWxsPSIjQzMwMDJGIiBkPSJNMTI1IDMwdjIyLjItLjFWMjMwbDc4LjktNDMuNyAxNC4yLTEyMy4xTDEyNSAzMHoiIC8+CiAgICA8cGF0aCAgZmlsbD0iI0ZGRkZGRiIgZD0iTTEyNSA1Mi4xTDY2LjggMTgyLjZoMjEuN2wxMS43LTI5LjJoNDkuNGwxMS43IDI5LjJIMTgzTDEyNSA1Mi4xem0xNyA4My4zaC0zNGwxNy00MC45IDE3IDQwLjl6IiAvPgogIDwvc3ZnPg=="]');
    expect(element).toBeTruthy();
  });

  test('should have another title in a h2 tag', () => {
    expect(componentDomDebug.nativeElement.querySelector('h2').textContent).toContain('List of links');
  });

  test('should have a unordered list of links', () => {
    const ul = componentDomDebug.query(By.css('ul'));
    expect(ul).toBeTruthy();

    expect(ul
      .query(By.css('li:nth-child(1)'))
      .query(By.css('button'))
      .nativeElement.textContent
    )
    .toContain('Click me to add');

    expect(ul
      .query(By.css('li:nth-child(2)'))
      .query(By.css('h2'))
      .query(By.css('nav'))
      .query(By.css('a'))
      .nativeElement.getAttribute('routerLink')
    )
    .toBe('/users');

    expect(ul
      .query(By.css('li:nth-child(3)'))
      .query(By.css('h2'))
      .query(By.css('nav'))
      .query(By.css('a'))
      .nativeElement.getAttribute('routerLink')
    )
    .toBe('/users-table');
  });

  test('should have method click which adds someNumber', async(() => {
    // spyOn(component, 'click');
    // fixture.whenStable().then(() => {
    //   expect(component.click).toHaveBeenCalled();
    //   expect(component.someNumber).toBe(2);
    // });
    component.click();
    expect(component.someNumber).toBe(2);
    component.click();
    expect(component.someNumber).toBe(3);
  }));

});
