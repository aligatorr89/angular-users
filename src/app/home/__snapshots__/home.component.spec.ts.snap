// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render correctly 1`] = `
<app-home
  someNumber={[Function Number]}
  title={[Function String]}
>
  <div
    style="text-align:center"
  >
    <h1>
       Welcome to angular-users! 
    </h1>
    <img
      alt="Angular Logo"
      src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNTAgMjUwIj4KICAgIDxwYXRoIGZpbGw9IiNERDAwMzEiIGQ9Ik0xMjUgMzBMMzEuOSA2My4ybDE0LjIgMTIzLjFMMTI1IDIzMGw3OC45LTQzLjcgMTQuMi0xMjMuMXoiIC8+CiAgICA8cGF0aCBmaWxsPSIjQzMwMDJGIiBkPSJNMTI1IDMwdjIyLjItLjFWMjMwbDc4LjktNDMuNyAxNC4yLTEyMy4xTDEyNSAzMHoiIC8+CiAgICA8cGF0aCAgZmlsbD0iI0ZGRkZGRiIgZD0iTTEyNSA1Mi4xTDY2LjggMTgyLjZoMjEuN2wxMS43LTI5LjJoNDkuNGwxMS43IDI5LjJIMTgzTDEyNSA1Mi4xem0xNyA4My4zaC0zNGwxNy00MC45IDE3IDQwLjl6IiAvPgogIDwvc3ZnPg=="
      width="300"
    />
  </div><h2>
    List of links
  </h2><ul>
    <li>
      <button
        name="button"
        type="button"
      >
        Click me to add
      </button>
    </li>
    <li>
      <h2>
        <nav>
          <a
            routerlink="/users"
          >
            USERS
          </a>
        </nav>
      </h2>
    </li>
  </ul>
</app-home>
`;
